core/src/test/java/org/apache/iceberg/TestPartitionSpecParser.java

```diff
@@ -40,26 +55,24 @@ public void testToJsonForV1Table() {
         "}";
     Assert.assertEquals(expected, PartitionSpecParser.toJson(table.spec(), true));
 
-    PartitionSpec spec = PartitionSpec.builderFor(table.schema())
-        .bucket("id", 8)
-        .bucket("data", 16)
-        .build();
-
-    table.ops().commit(table.ops().current(), table.ops().current().updatePartitionSpec(spec));
+    table.updateSpec().clear()
+        .addBucketField("id", 8)
+        .addBucketField("data", 16)
+        .commit();
```

Comment:
Why does this test use a table? It would make more sense to me if it created a custom partition spec (using add to set field IDs directly) and checked the resulting JSON string, without using update or a table at all.